src/sfdt/toggleBookmark.ts

```diff
@@ -5,6 +5,88 @@ import {
 	isBookmarkStart,
 	isBookmarkEnd,
 } from '../queryBookmark'
+import Stack from '../stack'
+
+const makeToggleOff = (inlines, name) => {
+  const newInlines = []; // Should act as queue for getting new list of inlines
+  const stackOfBookmarks = new Stack(); // will be used to find relationship of bookmarks
+  let toggleOffAtIndex = null;
+
+  inlines.forEach((inline, index) => {
+    const nextInline = inlines[index + 1];
+    const prevInline = inlines[index - 1];
+
+    if (!(inline.fieldType || inline.hasFieldEnd)) {
+      newInlines.push(inline);
+    }
+
+    if (inline.name && inline.name.split("::").includes("COND")) {
+      const isSameBookmark = isMatchingBookmark(inline, name);
+      if (isBookmarkStart(inline)) {
+        stackOfBookmarks.push(inline);
+
+        if (isSameBookmark) {
+          if (toggleOffAtIndex === null) {
+            toggleOffAtIndex = stackOfBookmarks.peekAt();
+          } else if (stackOfBookmarks.peekAt() <= toggleOffAtIndex) {
+            toggleOffAtIndex = stackOfBookmarks.peekAt();
+          }
+        }
+
+        if (inline.hasFieldEnd || inline.fieldType === 2) {
```

Comment:
same point here, let's try and remove this type of inline type checking, and have it in a named function instead, eg:

`const isHiddenAreaStartTag = (inline) => ...`
`const isHiddenAreaClosingTag = (inline) => ...`